src/pydargs/__init__.py

```diff
@@ -111,15 +131,22 @@ def _create_parser(tp: Type[Dataclass], **kwargs: Any) -> ArgumentParser:
             elif origin in UNION_TYPES:
                 union_parser = partial(_parse_union, union_type=field.type)
                 setattr(union_parser, "__name__", repr(field.type))
-                parser.add_argument(
+                parser_or_group.add_argument(
                     *arguments,
                     type=union_parser,
                     **argument_kwargs,
                 )
             else:
                 raise NotImplementedError(f"Parsing into type {origin} is not implemented.")
+        elif hasattr(field.type, "__dataclass_fields__"):
+            if positional:
+                raise ValueError("Dataclasses may not be positional arguments.")
+            if field.default_factory is not None and field.default_factory != field.type:
+                warn(f"Non-standard default of field {field.name} is ignored by pydargs.", UserWarning)
+            _add_arguments(parser, field.type, prefix=f"{prefix}{field.name}_")
```

Comment:
```suggestion
            # Recursively add arguments for the nested dataclasses
            _add_arguments(parser, field.type, prefix=f"{prefix}{field.name}_")
```
I think this makes it easier to spot the recursion in the code?